### PR TITLE
Draw perfect subtrees as a single element when they're large

### PR DESCRIPTION
**Pull request by AlCutter**
Adds the ability for `treetex` to render very large Merkle trees by coalescing large perfect subtrees into a single element, like this:

texput

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by Martin2112** [comment]
This is mega!

File: docs/merkletree/treetex/main.go [hunk 1]
```diff
@@ -125,6 +140,21 @@ func modifyNodeInfo(k string, f func(*nodeInfo)) {
 	nInfo[k] = n
 }
 
+// perfectMega renders a large perfect subtree as a single entity.
+func perfectMega(prefix string, height, leafIndex int64) {
+	stLeaves := int64(1 << uint(height))
```
Comment (pavelkalinnikov) [inline docs/merkletree/treetex/main.go]: `int64(1) << uint(height)`

File: docs/merkletree/treetex/main.go [hunk 2]
```diff
@@ -125,6 +140,21 @@ func modifyNodeInfo(k string, f func(*nodeInfo)) {
...
+	stWidth := float32(stLeaves) / float32(*treeSize)
+	fmt.Printf("%s [%d\\dots%d, edge label={node[midway, above]{%d}}, perfect, tier=leaf, minimum width=%f\\linewidth ]\n", prefix, leafIndex, leafIndex+stLeaves, stLeaves, stWidth)
+
+	// Create some hidden nodes to preseve the tier spacings:
+	for i := height - 2; i > 0; i-- {
+		fmt.Printf(" [, no edge, tier=%d ", i)
+	}
+	for i := height - 2; i > 0; i-- {
+		fmt.Printf(" ] ")
```
Comment (pavelkalinnikov) [inline docs/merkletree/treetex/main.go]: could nest this to the previous loop with a `defer` expression

File: docs/merkletree/treetex/main.go (hunk 1 above)
Comment (AlCutter) [reply docs/merkletree/treetex/main.go]: #1445 fixes this

File: docs/merkletree/treetex/main.go (hunk 2 above)
Comment (AlCutter) [reply docs/merkletree/treetex/main.go]: #1445 fixes this

**Pull request merged**